### PR TITLE
update initData.sh

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 1]
```diff
@@ -394,6 +394,23 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
       return assetService.getAssetMetricsDataById(assetID, starttime, duration);
    }
 
+   @ResponseStatus(HttpStatus.OK)
+   @RequestMapping(value = "/{id}/metrics", method = RequestMethod.GET)
+   public List<MetricData> getAssetMetricsDataDuration(@PathVariable("id") String assetID,
+         @RequestParam(value = "starttime", required = false) Long starttime,
```
Comment (Pengpengwanga) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java:400]: getMetrics

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 2]
```diff
@@ -394,6 +394,23 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
...
+         starttime = System.currentTimeMillis() - FIVE_MINUTES;
+      }
+      if (duration == null || duration <= 0) {
+         duration = FIVE_MINUTES;
+      }
+      if (duration > FIVE_MINUTES * 24) {
+         duration = FIVE_MINUTES * 24;
+      }
+      return assetService.getAssetDurationMetricsDataById(assetID, starttime, duration);
+   }
```
Comment (Pengpengwanga) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java:412]: getMetricsByID

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 3]
```diff
@@ -394,6 +396,23 @@ public void insertRealtimeData(@PathVariable("id") String assetID,
...
+   public List<MetricData> getMetrics(@PathVariable("id") String assetID,
+         @RequestParam(value = "starttime", required = false) Long starttime,
+         @RequestParam(value = "duration", required = false) Integer duration) {
+      if (starttime == null || starttime <= 0) {
+         starttime = System.currentTimeMillis() - FIVE_MINUTES;
+      }
+      if (duration == null || duration <= 0) {
+         duration = FIVE_MINUTES;
+      }
+      if (duration > FIVE_MINUTES * 24) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java:410]: 1 hour

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 4]
```diff
@@ -877,4 +878,67 @@ private String getSensorPositionInfo(Asset asset) {
       }
       return positionInfo.toString();
    }
+
+   public List<MetricData> getMetricsByID(String assetID, Long starttime, Integer duration){
+      Map<AssetCategory, List<MetricData>> categoryAndMetricDataMap =
+            getAllMetricsByIDAndCategory(assetID, starttime, duration);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:884]: remove category

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 5]
```diff
@@ -877,4 +878,67 @@ private String getSensorPositionInfo(Asset asset) {
...
+      if(categoryAndMetricDataMap.isEmpty()) {
+         return null;
+      }
+      return filterAndGetAllMetricData(categoryAndMetricDataMap);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:888]: removeAggregatedMetrics

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 6]
```diff
@@ -877,4 +878,635 @@ private String getSensorPositionInfo(Asset asset) {
...
+      Map<String, String> hostMetricsFormula =
+            server.metricsFormulaToMap(hostMetricsFormulaString, new TypeReference<Map<String, String>>() {});
+      Map<String, List<String>> assetIdAndMetricsNameList = Maps.newHashMap();
+      for (Map.Entry<String, String> itemEntry : hostMetricsFormula.entrySet()) {
+         List<String> metricsNameList = assetIdAndMetricsNameList.computeIfAbsent(itemEntry.getValue(), k -> Lists.newArrayList());
+         metricsNameList.add(itemEntry.getKey());
+      }
+      for (Map.Entry<String, List<String>> entry : assetIdAndMetricsNameList.entrySet()) {
+         String assetID = entry.getKey();
+         Asset asset = getAssetById(assetID);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:1024]: add check.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 7]
```diff
@@ -877,4 +878,635 @@ private String getSensorPositionInfo(Asset asset) {
...
+         Asset metricAsset = entry.getKey();
+         List<ValueUnit> valueUnits = entry.getValue();
+         if(AssetCategory.Server != metricAsset.getCategory()) {
+            continue;
+         }
+         Set<String> specialMetricNames = new HashSet<String>();
+         specialMetricNames.add(MetricName.SERVER_AVERAGE_USED_POWER);
+         specialMetricNames.add(MetricName.SERVER_PEAK_USED_POWER);
+         specialMetricNames.add(MetricName.SERVER_MINIMUM_USED_POWER);
+         specialMetricNames.add(MetricName.SERVER_ENERGY_CONSUMPTION);
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:1047]: double check. energy metric. overlap.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 8]
```diff
@@ -877,4 +878,635 @@ private String getSensorPositionInfo(Asset asset) {
...
+         for(ValueUnit value : realTimeData.getValues()) {
+            valueunits.add(value);
+         }
+      }
+      return valueunits;
+   }
+
+   private void getSensorValueUnit(Map<Asset, List<ValueUnit>> assetAndValueUnitsMap,
+         long starttime, int duration, Map<String,String> locationAndIdMap, String metricName){
+      for(Map.Entry<String, String> locationInfoAndId : locationAndIdMap.entrySet()) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:1489]: refactor logic. get assetid, then get value unit for each asset.

**Review by yixingjia — APPROVED** [review]

**Pull request merged**